libc/cmake/modules/LLVMLibCArchitectures.cmake

```diff
@@ -69,6 +69,10 @@ function(get_arch_and_system_from_triple triple arch_var sys_var)
     set(target_sys "darwin")
   endif()
 
+  if(target_sys STREQUAL "unknown")
+    list(GET triple_comps -1 target_sys)
+  endif()
```

Comment:
I'm not sure if this is completely valid since someone could use `aarch64-unknown-uefi-llvm` as a valid triple to describe this toolchain.

Reply:
Probably making it 2 instead of -1 would work for that. I just tested it with 2 using `aarch64-unknown-uefi` and nothing broke.

Reply:
`uefi` in this case should pretty easily replace `linux`, can we not use the same handling?

Reply:
`aarch64-unknown-linux-llvm` and `aarch64-unknown-uefi-llvm` should be able to use the exact same parsing.

Reply:
Yes, since they both match on unknown. It would just shift to the 3rd component.

Reply:
I'm just trying to figure out why this is necessary for the UEFI case but not the Linux case.

Reply:
I'm not sure either, it was setting `LLVM_TARGET_OS` to unknown before the change.

Reply:
Maybe the lack of the ABI in the triple freaks it out? Man I hate github code reviews, can't even comment on related lines of code.

```
  string(REPLACE "-" ";" triple_comps ${triple})
  list(LENGTH triple_comps triple_size)
  if(triple_size LESS "3")
    return()
    return()
  endif()
  math(EXPR system_index "${triple_size} - 2")
  list(GET triple_comps 0 target_arch)
```
Likely it just gets confused with a triple of size three and indexes into the system. Honestly it should probably just take the third element here

Reply:
Oh that's possible, though it probably is still good to have this change. Maybe we should rework that function at a later time to properly accept doubles (`$arch-$os`), triple (`$arch-$vendor-$os`), and the full tuple (`$arch-$vendor-$os-$abi`). The function feels a little over engineered.